### PR TITLE
Initial setup code review, lint and coding guidelines

### LINKED ISSUE #302: Bring Review Guidelines, Code Style to this repository
We want the Android repository to have equivalent Guidelines as the ones defined for iOS and added on this PR https://github.com/zcash/secant-ios-wallet/pull/24

- [x] Linting
- [x] Code Review Guidelines (these are pretty general)
- [x] Bug report, ux-report,  feature request and pull request templates
- [x] Readme
- [x] Docs folder
- [x] Docs testing folder

### PR DESCRIPTION
**Pull request by pacu**
Closes #302 

- bug report template
- feature request template
- ux report template
- Architecture readme
- swiftlint according to secant wallet
- local coverage information
- manual testing documentation

NOTE: this PR will might break CI because introduces changes on the existing linting rules. This issues will be resolved in a follow up commit

### CODE REVIEW

File: SWIFTLINT.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+This violates the rule that you should not use trailing closure syntax when a method accepts multiple closure parameters, so SwiftLint will flag it with a warning. 
+
+You can work around this by extracting the `Button`'s action into a separate method. While this is frequently a better solution when the action requires several statements, it's an onerous requirement when the action is a single statement, as in the example above.
+
+In these cases, you're permitted to disable this rule **for the declaration of a SwiftUI view** only. The rule name is `multiple_closures_with_trailing_closure`.
+
+### Open-source files
+
+Occasionally, you'll find it necessary to include an unmodified open-source file in the sample project. It's a virtual certainty that these files won't comply with our style guide. Our practice has always been to leave these files in their original state. In this situation, you should disable SwiftLint for the entire file:
```
Comment (adamstener) [inline SWIFTLINT.md:126]: I think there was a comment on the other PR about pulling this section, unless I misunderstood the intention there.

File: .swiftlint.yml [hunk 2]
```diff
@@ -1,45 +1,144 @@
...
+  
+private_over_fileprivate:
+  validate_extensions: true
+
+trailing_whitespace:
+  ignores_empty_lines: true
+  ignores_comments: true
+
+vertical_whitespace:
+  max_empty_lines: 2
```
Comment (adamstener) [inline .swiftlint.yml:144]: There was a discussion in the previous PR to move this to 1 for consistency.  I don't think that change was made prior to merging, however the PR that added the change to include enabling SwiftLint was written to follow a value of 1 here and tested as such.  I think the concern there was pragma marks or comments wanting to have the ability to have whitespace around them and that does in fact work with a value of 1 here, so we should be safe to make that change in both repos.

File: CODE_REVIEW_GUIDELINES.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+#  Specification:
+
+- Are the requirements for the change well specified? Where are they documented? Bugfixes need a clear specification of the bug's cause and fix. (Small PRs might specify in a github ticket. Large changes may require stand-alone docs, or ZIPs.)
+- Is there a documented test plan, which describes how to manually verify the change works on testnet prior to a release?
+
+# User Documentation:
+
+- What is the "changelog" entry for this change? (All changes should include this.)
+- Are there any changes which require updates to user-facing documentation? If so, does the new documentation make sense? (documentation should be placed in [docs/](/docs) folder of the repo)
```
[the remark this answers is not in the document]
Comment (pacu) [reply CODE_REVIEW_GUIDELINES.md:15]: That's right. Probably "Documentation" it's for the better

File: CODE_REVIEW_GUIDELINES.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+# Testing:
+
+- For non-minor PRs (up to the code reviewers and PR creator if this is needed), we require authors to perform a thorough self-review and self-test of the resulting code base, including use cases that might not be visible affected by the introduced changes. Reviewers are not expected to run the changes locally, but are definitely encouraged to do so at their best judgement.
```
[the remark this answers is not in the document]
Comment (pacu) [reply CODE_REVIEW_GUIDELINES.md:19]: I'll change this on both places

File: SWIFTLINT.md (hunk 1 above)
Comment (pacu) [reply SWIFTLINT.md:126]: I'll change this on both places. thanks

File: .swiftlint.yml (hunk 2 above)
Comment (pacu) [reply .swiftlint.yml:144]: good, I'll add this change with the rest of the work. I still need to run tests to see that nothing as been broken. there's a rule about enforcing the use of `isEmpty` that inverts the boolean logic of some guard statements that are sensitive.

Comment (adamstener) [reply .swiftlint.yml:144]: Ah that's a good catch.  If that one causes any issues at all I don't think it would be controversial at all to remove.

**Pull request merged**